### PR TITLE
Add StringSyntaxAttribute.Json

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/runtime/issues/62505#issuecomment-993866275
cc: @eiriktsarpalis, @danmoseley, @CyrusNajmabadi

### CODE REVIEW

**Review by eiriktsarpalis — APPROVED** [review]
Thanks!

File: src/libraries/System.Text.Json/ref/System.Text.Json.cs [hunk 1]
```diff
@@ -193,11 +193,11 @@ public static partial class JsonSerializer
         public static object? Deserialize(System.ReadOnlySpan<byte> utf8Json, System.Type returnType, System.Text.Json.JsonSerializerOptions? options = null) { throw null; }
         public static object? Deserialize(System.ReadOnlySpan<byte> utf8Json, System.Type returnType, System.Text.Json.Serialization.JsonSerializerContext context) { throw null; }
```
Comment (stephentoub) [inline src/libraries/System.Text.Json/ref/System.Text.Json.cs:194]: If/when C# supports utf8 literals, we'll presumably want to attribute these utf8json parameters as well.

**Comment by danmoseley** [comment]
Just curious @CyrusNajmabadi do you imagine these annotations could possibly influence the debugger views, not just colorization/squiggling? I'm thinking eg the selection of watch window visualizer perhaps.

**Comment by CyrusNajmabadi** [comment]
Sorry, just got around to this.  One sneaky things here is that really are multiple Jsons.  I presume if we see this we should think this is our json, not Json.net right (roslyn supports both) :)

> Just curious @CyrusNajmabadi do you imagine these annotations could possibly influence the debugger views, not just colorization/squiggling? I'm thinking eg the selection of watch window visualizer perhaps.

That seem like a great idea.  Though would likely need to go through the debugger stack here.  That said, visualizers are ancient and haven't been updated in what feels like forever (and they're still model, etc.).  I would not hold my breath on that unless there is a renewed interest in updating them.

**Pull request merged**